### ISSUE TITLE
Include TypeScript Lessons as part of the freeCodeCamp Curriculum

### ISSUE THREAD

**Issue by RJamesMaier** [freeCodeCamp/freeCodeCamp#39828]
**Idea**
According to the most recent [stack overflow](https://insights.stackoverflow.com/survey/2020#most-loved-dreaded-and-wanted) survey, TypeScript is among the most loved languages and is becoming a more popular way of writing JavaScript.

**Solution**
TypeScript, particularly when compared to JavaScript, will probably be a bit much for a beginner, but perhaps in an "advanced" section, for those who want to learn further, the curriculum could include TypeScript.

**Additional context**
I would love to work on a curriculum for learning TypeScript, as I myself know what it is like, to have to pull in different resources to begin to understand TypeScript. With the freeCodeCamp platform, other developers who also want to have a look at TypeScript do not have to spend their time, searching for different resources.

**Comment by ojeytonwilliams** [comment]
Hi @RJamesMaier, it would be great if we could offer TypeScript lessons, but there's a bit of setup we'd need first.  While it should not be too difficult, since there's a [proof of concept](https://www.typescriptlang.org/play), we still have a bit of work left to do on the editor before we can extend it to cover TS.

Also, @RandellDawson is best placed to advise about how (and if!) we might go about fitting this into the curriculum.

To summarise, in principle this is a great idea, but we have a few things to work out before we move forward.

**Comment by RandellDawson** [comment]
@RJamesMaier Our current goal is to include at least one practice project to teach the basics of TypeScript in the v7 curriculum.

**Comment by moT01** [comment]
Thank you for reporting this issue.

This is a standard message notifying you that the issue has become stale, so we are going to close it.

If you think we're wrong in closing this issue, please request for it to be reopened. Thank you and happy coding.
